src/channel/NodeChannel.js

```diff
@@ -13,15 +14,36 @@ export default class NodeChannel extends Channel {
     /**
      * @internal
      * @param {string} address
+     * @param {string | undefined} certificate
      */
-    constructor(address) {
+    constructor(address, certificate) {
         super();
 
+        let clientCredentials;
+        let clientAddress;
+
+        if (certificate != null) {
+            clientCredentials = credentials.createSsl(
+                Buffer.from(utf8.encode(certificate))
+            );
+            clientAddress = `${address.split(":")[0]}:50212`;
```

Comment:
I don't think we should be doing this. If someone is running their own version of the network, and they've enabled TLS on a different port then this will fail for no reason.